mlflow/models/model.py

```diff
@@ -1393,3 +1397,23 @@ def set_model(model) -> None:
             pass
 
     raise mlflow.MlflowException(SET_MODEL_ERROR)
+
+
+def _update_active_model_id_based_on_mlflow_model(mlflow_model: Model):
+    """
+    Update the current active model ID based on the provided MLflow model.
+    Only set the active model ID if it is not already set by the user.
+    This is useful for setting the active model ID when loading a model
+    to ensure traces generated are associated with the loaded model.
+    """
+    if (
+        mlflow_model.model_id
+        and (amc := _get_active_model_context())
+        # only set the active model if the model is not set by the user
+        and not amc.set_by_user
+        and amc.model_id != mlflow_model.model_id
+    ):
+        _set_active_model_id(model_id=mlflow_model.model_id)
+        _logger.info(
+            "Use `mlflow.set_active_model` to set the active model to a different one if needed."
+        )
```

Comment:
```suggestion
    if mlflow_model.model_id is None:
        return
   
    active_model = _get_active_model_context())
    if (
        active_model.model_id != mlflow_model.model_id
        # only set the active model if the model is not set by the user
        and not active_model.set_by_user
     ):
         _set_active_model_id(model_id=mlflow_model.model_id)
```

nit: The `if` condition is too complex to read imo, let's flatten it.